crates/optimism/primitives/src/supervisor/traits.rs

```diff
@@ -0,0 +1,73 @@
+//! RPC validator component used in interop.
+//!
+//! [`InteropTxValidator`] parses inbox entries from [`AccessListItem`]s, and queries a
+//! superchain supervisor for their validity via RPC using the [`CheckAccessList`] API.
+
+use crate::supervisor::{
+    parse_access_list_items_to_inbox_entries, CheckAccessList, ExecutingDescriptor,
+    InteropTxValidatorError, SafetyLevel,
+};
+use alloc::boxed::Box;
+use alloy_eips::eip2930::AccessListItem;
+use alloy_primitives::B256;
+use core::time::Duration;
+
+/// Interacts with a Supervisor to validate inbox entries extracted from [`AccessListItem`]s.
+#[async_trait::async_trait]
+pub trait InteropTxValidator {
```

Comment:
do we ever expect that validation won't require an rpc request?

if not all of this logic can just be moved into the client type as native functions.